### PR TITLE
#164798009 setup testing for the app

### PR DESCRIPTION
**Pull request by vic3king**
#### What does this PR do?
Sets up unit testing.
#### Description of Task to be completed?
- install dependencies nyc,mocha,chai,chai-http
- setup testing
- add scripts
#### How should this be manually tested?
clone the repo https://github.com/andela/hermes-ah-backend
cd into the repo
npm run test on the terminal
#### Any background context you want to provide?
NA
#### What are the relevant pivotal tracker stories?
[164798009](https://www.pivotaltracker.com/n/projects/2318856/stories/164798009)
#### Screenshots
image

#### Questions:
NA

### CODE REVIEW

File: tests/index.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      .request(app)
+      .get('/')
+      .end((err, res) => {
+        expect(res).to.have.status(200);
+        done();
+      });
+  });
+});
+
+export { chai, expect, app };
```
Comment (rotimi-babalola) [inline tests/index.test.js]: Why are you exporting `chai`, `expect` and `app`?

File: tests/index.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable no-undef */
```
Comment (rotimi-babalola) [inline tests/index.test.js]: Why did you need to disable this rule?

**Review by rotimi-babalola — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "migrate:undo": "sequelize db:migrate:undo:all",
     "seed": "sequelize db:seed:all",
     "start:dev": "npm run migrate && nodemon server/app.js --exec babel-node",
+    "test": "export NODE_ENV=test || SET \"NODE_ENV=test\" && nyc mocha tests/**/*.test.js --timeout 20000 --require @babel/register --exit",
```
Comment (rotimi-babalola) [inline package.json]: What do you need the `SET` command for?

**Comment by rotimi-babalola** [comment]
Also resolve conflicts @vic3king

File: tests/index.test.js (hunk 1 above)
Comment (vic3king) [reply tests/index.test.js]: we have an index file for our test. I exported chai, expect and the app so for each of our test files we will import from the index file and use it. That was the setup we tried to explain earlier today explained

File: tests/index.test.js (hunk 2 above)
Comment (vic3king) [reply tests/index.test.js]: Tha was an error on my part, I Have fixed that.

File: tests/index.test.js (hunk 1 above)
Comment (rotimi-babalola) [reply tests/index.test.js]: @vic3king But you don't need to export `chai`, `expect` and `app` from this file. If they are needed in any other test file we can easily import it directly can't we? Like so:

```
import chai, { expect } from 'chai';
import app from './pathToApp.js
```

I think the use of the index file in your scenario is useful when you have a bunch of functions from different files that you want to export from one file.

In the case of our tests, I don't think you need an index file for the tests instead you could have different tests to test different parts of the app e.g. server, controllers, etc and run all those tests with the `npm test` command

Do you understand?

File: package.json (hunk 3 above)
Comment (vic3king) [reply package.json]: Coming from a windows OS background I normally would have that on my tests script to enable me to set my test env. export works for mac/linux

Comment (rotimi-babalola) [reply package.json]: Since we are working with mac/linux environments (both locally and for testing) you can remove this command

File: tests/index.test.js (hunk 1 above)
Comment (vic3king) [reply tests/index.test.js]: Yes, I understand that. I'll make the necessary correction.

File: package.json (hunk 3 above)
Comment (vic3king) [reply package.json]: Will do.

**Comment by rotimi-babalola** [comment]
Once the PR for fixing houndCI is merged we can merge this

**Comment by rotimi-babalola** [comment]
Please update this branch @vic3king

**Comment by vic3king** [comment]
> Please update this branch @vic3king
Done, I had to rebase.

**Comment by rotimi-babalola** [comment]
I will merge this but I still don't know why HoundCI is still complaining
